src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php

```diff
@@ -130,11 +130,11 @@ public function reverseTransform($value)
         try {
             if ($dateOnly) {
                 // we only care about year-month-date, which has been delivered as a timestamp pointing to UTC midnight
-                return new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->inputTimezone));
+                $dateTime = new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->outputTimezone));
```

Comment:
wait, @tkleinhakisa is right! this should be inputTimezone instead of outputTimezone!
when dateOnly is true, the DateTime should not depend on the current *hour*!